truss/templates/server/model_wrapper.py

```diff
@@ -777,6 +787,20 @@ async def chat_completions(
         )
         return await self._process_model_fn(inputs, request, descriptor=descriptor)
 
+    async def websocket(self, ws: WebSocket) -> None:
+        # FastAPI automatically completes the upgrade request for websockets, so we can't
+        # return a helpful HTTP error to the client. Sending an error message before we
+        # explicitly close the connection is the better option.
+        descriptor = self.model_descriptor.websocket
+        if not descriptor:
+            await ws.send_text(f"{MethodName.WEBSOCKET} not implemented.")
+            await ws.close(code=1003)  # 1003 = Unsupported
+            return
+
+        if descriptor.is_async:
```

Comment:
We have two common patterns for actually invoking the model function:
- If it's the more 'typical' path, where we want to parse the request body, handle the response, we go through this [function](https://github.com/basetenlabs/truss/blob/057409d6c527016e745d53e89798169c01c69ac2/truss/templates/server/model_wrapper.py?plain=1#L701-L714).
- We now have a couple of functions (health checks, setup environment, and now websocket) that have a very similar structure, but combining with above didn't actually make things cleaner, mainly since the arguments are all different. Thought about introducing a separate type of helper that deals with these exceptions, but then I thought it was confusing to decide which one to use. For now I left separate - but open to suggestions here!